ivy/functional/frontends/numpy/mathematical_functions/extrema_finding.py

```diff
@@ -51,3 +50,41 @@ def amin(
         a = ivy.concat([a, header], axis=axis)
 
     return ivy.min(a, axis=axis, keepdims=keepdims, out=out)
+
+
+def nanmin(
+    a,
+    axis=None,
+    out=None,
+    keepdims=False,
+    initial=None,
+    where=True,
+):
+    nan_mask = ivy.isnan(a)
+    a = ivy.where(ivy.logical_not(nan_mask), a, ivy.default(out, a.full_like(+ivy.inf)))
+    if initial is not None:
+        if ivy.is_array(where):
+            a = ivy.where(where, a, ivy.default(out, a.full_like(initial)))
+            nan_mask = ivy.where(where, nan_mask, nan_mask.full_like(True))
+        s = ivy.shape(a, as_array=True)
+        if axis is not None:
+            if isinstance(axis, (tuple, list)) or ivy.is_array(axis):
+                # introducing the initial in one dimension is enough
+                ax = axis[0] % len(s)
+                s[ax] = 1
+            else:
+                ax = axis % len(s)
+                s[ax] = 1
+        header = ivy.full(ivy.Shape(s.to_list()), initial, dtype=ivy.dtype(a))
+        if axis:
+            if isinstance(axis, (tuple, list)) or ivy.is_array(axis):
+                a = ivy.concat([a, header], axis=axis[0])
+            else:
+                a = ivy.concat([a, header], axis=axis)
+        else:
+            a = ivy.concat([a, header], axis=0)
+    res = ivy.min(a, axis=axis, keepdims=keepdims, out=out)
+    mask = ivy.all(nan_mask, axis=axis, keepdims=keepdims)
+    if ivy.any(mask):
+        res = ivy.where(ivy.logical_not(mask), res, ivy.nan, out=out)
```

Comment:
As I see it, you are reintroducing the nan values into the res array and thus these lines should be removed, let me know if I'm missing something!

Reply:
This is standard numpy operation, where if all the values in the axes is masked out by where, the axis should be nan.

Reply:
Yeah but you are overriding the nan_mask with both the nan values and the where array, so an input like this diverges from the behaviour in `np.nanmin`

```
a = ivy.array([[ivy.nan, ivy.nan, -3.], [1., ivy.nan, 3.]])
where = ivy.array([[True, False, False], [True, True, True]])
keepdims = True
initial = 3.
axis = 1
```

Reply:
I think it should be 

```
mask = ivy.all(ivy.logical_not(where), axis=axis, keepdims=keepdims)
if ivy.any(mask):
    res = ivy.where(ivy.logical_not(mask), res, ivy.nan, out=out)
```

Reply:
Oh, I masked the nan mask itself. Thanks!!